projecttacoma/measure-repository
# Release draft

# Summary
This PR adds [Release](https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#release) functionality to the Authoring Repository. The user can now "Release" a draft artifact to the central Measure Repository.

## New behavior
The user can now "Release" a draft artifact to the central Measure Repository from the Authoring Repository. When a draft artifact is released, the artifact `status` changes from `draft` to `active`. The spec specifies the following for the Release operation: "To be released, an artifact is required to have a version specified." Therefore, when the user releases a draft, they are prompted to input a version. The draft cannot be released without a version specified. When the user adds a version and confirms the release, the active artifact will now appear in the Measure Repository and the draft artifact will be deleted from the Authoring Repository.

## Code changes
- `ReleaseModal.tsx` - Modal when the user clicks "Release" to confirm release of the draft artifact and also input for a version.
- `pages/authoring/[resourceType]/[id].tsx` - Add modal
- `app/package.json` - add luxon
- `service/package.json` - add cors
- `app.ts` - expose headers using cors

# Testing guidance
- `npm run check:all`
- `npm run start:all`
- Navigate to the Authoring tab. Make sure you have a couple draft Measures and Libraries.
- Navigate to one of the draft artifacts. Click "Release".
- Add a version and confirm release.
- Check the following things:
    - The draft artifact is no longer in the Authoring repository 
    - The resourceType count in the Authoring repository is decremented by one
    - The active artifact is now in the Measure repository
    - The resourceType count in the Measure repository is incremented by one

A few notes/questions for the reviewer:
- One of the comments/questions Lauren left when she worked on this PR was the following: "TODO/question: -> should we check service for whether artifact already exists in some way for PUT update?" I don't think we need this since the artifact IDs should theoretically all be the same? Would love some feedback on that though.
- Bree added some awesome UI for the modal when she worked on this, but it wasn't fully functional quite yet and in an effort to get this functionality up before the connectathon, I put her stuff on a separate branch (`release_draft_UI`) and just did some simple UI.
- I added some descriptions of what the release does in the Modal, but should I add descriptions to the README.md and/or the homepage of the app?
